### ISSUE TITLE
Harbor not able to pull images from Azure Conrainer Registry (ACR) using ACR tokens

### ISSUE THREAD

**Issue by humbirs** [goharbor/harbor#16888]
Hi Team

I have deployed Harbor '**v2.5.0-98e1b82f**'  deployed on Kubernetes version 1.21.7, on my environment and I am getting issue while pulling images from Azure Container registry using **ACR token**  which is a preview service, Also using `test connection`   button it show connection tested successfully, but actual replication pull is failing with error.

Testing via CLI on linux machine test was successful.  `echo $TOKEN_PWD | docker login --username $TOKEN_NAME --password-stdin ACRREGISTRY.azurecr.io`

However if I use ACR Access keys  I am able to pull (since this access key has Admin access over ACR, our team is reluctant to use it)

**Steps followed:**

- Logon to Azure portal
- Locate ACR
- From left hand menu go to "Repository permissions" -> Tokens (Preview)
- Create a token by clicking on +ADD and assign a scope _repositories_pull 
- Generate any password 
- Now we have the Token Name and its password
- Go to Harbor GUI and logon 
- Create a registry endpoint pointing to ACR and in Access ID and Access Secret use TOKEN NAME and TOKEN Password   
- Create a replication rule then click on replicate.
- Tested with full permission for scope used for token, scope as per below

**Scope map permissions are applicable to all repositories in your container registry.**

```
    metadata/read
    metadata/write
    content/read
    content/write
    content/delete
```

**Expected result:**  It should pull the repo and artifact
**Actual result:**  Job status is failed

`failed to fetch artifacts: filed to fetch artifacts: failed to list artifacts of repository <<repo name>>: http status code: 401, body: {""Errors":[{"code":"UNAUTHORIZED", "message": "authentication required, visit https://aka.ms/acr/authorization for more information "`

I had raised or discussed the same on SLACK #harbor [Forum Thread](https://cloud-native.slack.com/archives/CC1E09J6S/p1652355546302649)  channel and it was concluded that 

```
I have some discoveries, for standard docker client process, the scope’s action is pull/push, but for acr is metadata_read/…, so with incorrect action then acr return 401 when list tags.

Currently harbor does not support use token as acr access key, please summary our discussion and open an issue in the github, thanks~

```
**More details** 

API reference followed "spec(https://github.com/opencontainers/distribution-spec/blob/main/spec.md#endpoints),  `/v2/<name>/tags/list`" 
Tested ACR token via GIT API to fetch the tag list and it was successful using below steps:  (reference I took is from https://github.com/Azure/acr/blob/main/docs/Token-BasicAuth.md )

1. Encoded the TokenName and Token Password  `echo -n '[username]:[password]' | base64`
2. Copy the encoded value and set it as a environment variable ` export acr_credential="xxxxxxx" `
3. set the scope for pull push and metadata_read `export scope="repository:busybox:metadata_read,pull,push"`
4. set the registry variable `registry="XXXregistry.azurecr.io"`
5. Get the access token from azure via ->  `curl -v -H "Authorization: Basic $acr_credential" "https://$registry/oauth2/token?service=$registry&scope=$scope"`
7. Store the access token from previous output to a variable export `acr_access_token="eyJh .... das"`
8. Execute the API to get the list of tags `curl -v -H "Authorization: Bearer $acr_access_token" "https://$registry/v2/busybox/tags/list"`
9. From the result I can see the tags

**Comment by humbirs** [comment]
Hi Guys , 

Is this going to be targeted for any release? I am following it up from past months and seems like with this delay this product might loose its light. 

Thanks

**Comment by chlins** [comment]
> Hi Guys ,
> 
> Is this going to be targeted for any release? I am following it up from past months and seems like with this delay this product might loose its light.
> 
> Thanks

@humbirs Hi, this issue should be fixed in the harbor v2.6.x.
